### ISSUE TITLE
A distribution with the supplied name already exists.

### ISSUE THREAD

**Issue by Abhiroop25902** [microsoft/WSL#7998]
### Version

Microsoft Windows [Version 10.0.22543.1000]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.81.1

### Distro Version

Ubuntu and Kali-Linux

### Other Software

- Powershell
    - PSVersion= 5.1.22543.1000
- Windows Terminal from Microsoft Store
    - Version 1.11.3471.0 

### Repro Steps

Observed in these three commands, only tried with these three commands
- Open WSL Ubuntu: `wsl -d Ubuntu`
- Open WSL Kali-Linux: `wsl -d kali-linux`
- Open custom-made WSL instance: `wsl -d tensorflow`

### Expected Behavior

Prompt of the respective Distribution should turn on and their respective prompt be displayed on the Terminal

### Actual Behavior

The Distro fails to turn on and only one line of "error" is displayed: `A distribution with the supplied name already exists.`

### Diagnostic Logs

[Feedback Hub Share Link](https://aka.ms/AAfleg8)

**Comment by Biswa96** [comment]
Run `wsl -l -v` command to see already registered GNU/Linux distributions in WSL.

**Comment by Abhiroop25902** [comment]
Output of `wsl -l -v`
```
  NAME          STATE           VERSION
* Ubuntu        Stopped         2
  kali-linux    Stopped         2
  tensorflow    Stopped         2
```

**Comment by elsaco** [comment]
@Abhiroop25902 does it still happens if you do a `wsl --shutdown` first, then start a distro?

**Comment by Abhiroop25902** [comment]
@elsaco Tried `wsl --shutdown` and then starting a distro, same result. Even tried full Windows restart, same result

**Comment by elsaco** [comment]
Sorry to hear it didn't help. Please check what's in the registry at:

`hkcu\software\microsoft\windows\currentversion\lxss`

That's where all WSL instances are defined. Here's example of WSL on my system with ArchLinux, Fedora and Ubuntu:

```
PS C:\> reg query hkcu\software\microsoft\windows\currentversion\lxss

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss
    DefaultVersion    REG_DWORD    0x2
    DefaultDistribution    REG_SZ    {974f7502-2924-43c3-bba1-18032c9663f5}

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{3ec0f75d-865f-4f5f-9a09-69736f67469b}
HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{7cf79139-0ba4-4de5-9a68-89bb800e28b1}
HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{974f7502-2924-43c3-bba1-18032c9663f5}
```

And the name of each WSL instance:

```
PS C:\> reg query "hkcu\software\microsoft\windows\currentversion\lxss\{3ec0f75d-865f-4f5f-9a09-69736f67469b}" /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{3ec0f75d-865f-4f5f-9a09-69736f67469b}
    DistributionName    REG_SZ    Arch

PS C:\> reg query "hkcu\software\microsoft\windows\currentversion\lxss\{7cf79139-0ba4-4de5-9a68-89bb800e28b1}" /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{7cf79139-0ba4-4de5-9a68-89bb800e28b1}
    DistributionName    REG_SZ    Fedora

PS C:\> reg query "hkcu\software\microsoft\windows\currentversion\lxss\{974f7502-2924-43c3-bba1-18032c9663f5}" /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{974f7502-2924-43c3-bba1-18032c9663f5}
    DistributionName    REG_SZ    Ubuntu

```

**Comment by Abhiroop25902** [comment]
Same Three Distributions are coming up from the Registry Keys also, kali-linux, Ubuntu, and tensorflow
```
PS C:\> reg query hkcu\software\microsoft\windows\currentversion\lxss

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss
    DefaultVersion    REG_DWORD    0x2
    DefaultDistribution    REG_SZ    {c4d0486e-933d-442e-9cc9-deb7ad17b3ca}

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{0bc0524b-2bc6-4361-9cc1-fb4c03531ba9}
HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{c4d0486e-933d-442e-9cc9-deb7ad17b3ca}
HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{ef1ee1b1-fd7b-4f00-bb5e-797e663a5569}
PS C:\> reg query 'HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{0bc0524b-2bc6-4361-9cc1-fb4c03531ba9}' /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{0bc0524b-2bc6-4361-9cc1-fb4c03531ba9}
    DistributionName    REG_SZ    kali-linux

PS C:\> reg query 'HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{c4d0486e-933d-442e-9cc9-deb7ad17b3ca}' /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{c4d0486e-933d-442e-9cc9-deb7ad17b3ca}
    DistributionName    REG_SZ    Ubuntu

PS C:\> reg query 'HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{ef1ee1b1-fd7b-4f00-bb5e-797e663a5569}' /v DistributionName

HKEY_CURRENT_USER\software\microsoft\windows\currentversion\lxss\{ef1ee1b1-fd7b-4f00-bb5e-797e663a5569}
    DistributionName    REG_SZ    tensorflow

PS C:\>
```

**Comment by Abhiroop25902** [comment]
Tried to reinstall Ubuntu by uninstalling from store and using `wsl --install Ubuntu`, installation is successfully happening, but I am getting the following error when launching:
```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x800700b7
The distribution installation has become corrupted.
Please select Reset from App Settings or uninstall and reinstall the app.
Error: 0x800700b7 Cannot create a file when that file already exists.

Press any key to continue...

[process exited with code 1]
```

**Comment by Abhiroop25902** [comment]
Followed the error to [this](https://github.com/microsoft/WSL/issues/3352#issuecomment-402673907) comment and tried 
1. `sfc /scannow`
2. `dism /Online /Cleanup-Image /RestoreHealth`

Still the problem persists

**Comment by Abhiroop25902** [comment]
Here is the link to the feedback hub post I made: [feedback hub link](https://aka.ms/AAfleg8)

**Comment by OneBlue** [comment]
Thanks @Abhiroop25902.

Looking at the logs, this is the root cause of the issue you're seeing:
` 0x800700B7 | HcnCreateNetwork {"Success":false,"Error":"Cannot create a file when that file already exists. `

This is error comes from the HostNetworkingService, which is probably in a bad state. [Resetting HNS's state might help](https://juniper.github.io/contrail-windows-docs/Known_issues/HNS_error_bestiary/)

**Comment by Abhiroop25902** [comment]
Ran the "Level Alpha decontamination procedure" from the [guide to reset HNS's state](https://juniper.github.io/contrail-windows-docs/Known_issues/HNS_error_bestiary/) by @OneBlue 

My computer did have docker installed so only the following commands worked: 
1. `Get-NetNAT | Remove-NetNAT`
2. `Restart-Service hns`

This fixed the issue, every WSL Distro started without any problem

I did a couple of restarts, tried C code compilation and execution and tried WSLg and networking by running Wireshark. All executed flawlessly as before.

I can't thank you enough @OneBlue to help me fix this issue

I am closing this issue now as the issue has been solved.

Do I need to do something in the Feedback Hub also to close the feedback there?
